### DISCUSSION TITLE
What does quarkus.kubernetes-config.secrets actually do?

### DISCUSSION THREAD

**Discussion by jonsnowseven** [quarkusio/quarkus#44662] [Q&A] [answered]
Hello.

I am exploring the [`quarkus.kubernetes-config.secrets`](https://quarkus.io/guides/kubernetes-config#quarkus-kubernetes-config_quarkus-kubernetes-config-secrets) parameter.

* As I understood, Quarkus will fetch the secret values using K8s API (so I guess it will need the associated permissions) and inject them in the Quarkus Configuration? Am I assuming this correctly?
* If I have the k8s secret mapped to a volume, is there a way to inject the file contents in the Quarkus Config using any property?

Thank you in advance.

[a comment by an automated account is not shown]

**Comment by geoand** [comment, marked as answer]
> As I understood, Quarkus will fetch the secret values using K8s API (so I guess it will need the associated permissions) and inject them in the Quarkus Configuration? Am I assuming this correctly?

Absolutely correct!

> If I have the k8s secret mapped to a volume, is there a way to inject the file contents in the Quarkus Config using any property?

So you are asking how the application should be configured to use the file contents?

**Reply by jonsnowseven** [reply]
> So you are asking how the application should be configured to use the file contents?

Yes, precisely. Are there `quarkus.kubernetes.*` parameters to inject the file contents into the Quarkus Config (out-of-the-box, I mean)?

**Reply by geoand** [reply]
If you could configure the secret to inject the file as a `$PWD/config/application.properties`, it should just work

**Reply by jonsnowseven** [reply]
But besides that, is there a Quarkus-like way of doing it?

**Reply by geoand** [reply]
If you configure  `quarkus.kubernetes...` properly, you can set it up

**Reply by jonsnowseven** [reply]
Let's say I have a secret in /etc/secrets called `my-secret`. How should I set up Quarkus to inject the contents in the configuration?

**Reply by geoand** [reply]
I don't remember honestly as I have not tried it in a long time
